### PR TITLE
Jumbotron bg-focus updates

### PR DESCRIPTION
**Pull request by sebtrif**
Changes:
- jumbotron now supports bg-focus as a contextual background

Visual demo:
screen shot 2017-07-30 at 22 47 25

### CODE REVIEW

File: less/jumbotron.less [hunk 1]
```diff
@@ -9,6 +9,27 @@
...
 
+  &.bg-focus {
+    h1, .h1 {
+      color: @brand-navy;
+    }
+
+    .lead {
+      color: @brand-grey-plus-10;
+    }
+  }
```
Comment (StevePole) [inline less/jumbotron.less]: Don't think we need these two any more?

Comment (sebtrif) [reply less/jumbotron.less]: mmh not sure, i've added .bg focus as a way to keep the default jumbotron to it's current styles but just enforce a white bg

happy to move the default jumbotron styles to this then (plus migrate old styles onto navy bg + white font)

File: less/type.less [hunk 2]
```diff
@@ -347,18 +347,17 @@ strong {
...
 }
 .bg-focus {
-  background-color: #FFF;
+  background-color: @brand-white;
 }
 
-.jumbotron,
+
 .bg-primary {
   color: @brand-smoke-plus-20;
```
Comment (sebtrif) [inline less/type.less:355]: @StevePole in between this line and :374, which one do we actually use (i'd want to remove the one that is outdated if possible)

File: less/jumbotron.less (hunk 1 above)
Comment (StevePole) [reply less/jumbotron.less]: I think these are the default colours for these things anyway.  The only reason for having them is to overule the jumbotron styles, but over-ruling back to what they would have been anyway sucks.  That's why I was thinking we could move those text colours onto `.jumbotron-image` and `.hero-with-background` so we no longer have styles on jumbotron by default.

File: less/type.less (hunk 2 above)
Comment (StevePole) [reply less/type.less:355]: 374 looks incorrect.  We don't use the wash colour for bg-primary.

File: less/jumbotron.less [hunk 3]
```diff
@@ -5,13 +5,17 @@
...
 .jumbotron {
   padding: @jumbotron-padding 0;
-  color: @jumbotron-color;
-  background-color: @jumbotron-bg;
   transition: background-color 0.2s ease-in;
 
-  h1,
-  .h1 {
-    color: @jumbotron-heading-color;
+  &.bg-primary, &.bg-info, .jumbotron-image & {
```
Comment (StevePole) [inline less/jumbotron.less]: I don't think it's necessary to have bg-primary and bg-info here anymore.  Thos classes should change heading colours etc already.

**Review by StevePole — CHANGES_REQUESTED** [review]

File: less/jumbotron.less (hunk 3 above)
Comment (sebtrif) [reply less/jumbotron.less]: cool, i'll remove it and test 👍 
edit: leaving jumbotron-image in there, that one's actually needed to "simulate" bg-primary

**Review by StevePole — APPROVED** [review]
LGTM

**Pull request merged**